src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -657,3 +657,7 @@ void VM_Version::initialize_cpu_information(void) {
 
   _initialized = true;
 }
+
+int VM_Version::max_vector_size(BasicType bt) {
+  return MaxVectorSize / type2aelembytes(bt);
+}
```

Comment:
Please change to use the code in https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/cpu/aarch64/aarch64.ad#L2344 like what you have changed in x86 systems, together with the `Matcher::max_vector_size()`. Thanks a lot!